json/src/lib.rs

```diff
@@ -219,6 +219,66 @@ pub struct GetBlockHeaderResult {
     pub next_block_hash: Option<bitcoin::BlockHash>,
 }
 
+#[derive(Clone, PartialEq, Debug, Deserialize, Serialize)]
+pub struct GetBlockStatsResult {
+    #[serde(rename = "avgfee")]
+    pub avg_fee: u32,
+    #[serde(rename = "avgfeerate")]
+    pub avg_fee_rate: u32,
+    #[serde(rename = "avgtxsize")]
+    pub avg_tx_size: u32,
+    #[serde(rename = "blockhash")]
+    pub block_hash: bitcoin::BlockHash,
+    #[serde(rename = "feerate_percentiles")]
+    pub fee_rate_percentiles: FeeRatePercentiles,
+    pub height: u64,
+    pub ins: usize,
+    #[serde(rename = "maxfee")]
+    pub max_fee: u64,
+    #[serde(rename = "maxfeerate")]
+    pub max_fee_rate: u32,
+    #[serde(rename = "maxtxsize")]
+    pub max_tx_size: u32,
+    #[serde(rename = "medianfee")]
+    pub median_fee: u32,
+    #[serde(rename = "mediantime")]
+    pub median_time: u32,
+    #[serde(rename = "mediantxsize")]
+    pub median_tx_size: u32,
+    #[serde(rename = "minfee")]
+    pub min_fee: u32,
+    #[serde(rename = "minfeerate")]
+    pub min_fee_rate: u32,
+    #[serde(rename = "mintxsize")]
+    pub min_tx_size: u32,
+    pub outs: usize,
+    pub subsidy: u32,
+    #[serde(rename = "swtotal_size")]
+    pub sw_total_size: usize,
+    #[serde(rename = "swtotal_weight")]
+    pub sw_total_weight: usize,
+    #[serde(rename = "swtxs")]
+    pub sw_txs: usize,
+    pub time: u32,
+    pub total_out: usize,
+    pub total_size: usize,
+    pub total_weight: usize,
+    #[serde(rename = "totalfee")]
+    pub total_fee: u64,
```

Comment:
This should be a `bitcoin::Amount` (will also require `with = "bitcoin::util::amount::serde::as_sat"` in the derive).

Reply:
Oh yeah, true, we're using those in this crate.

Reply:
As far as I understand, bitcoin::Amount should then also be used for: `avg_fee`, `fee_rate_percentiles`, `max_fee`, `median_fee` and `min_fee`, right?

Reply:
When I try too define the fields as `Option<Amount>`, the compiler throws the follwoing error at me: _the trait `SerdeAmount` is not implemented for `std::option::Option<Amount>`_

Any idea how I should deal with this?

Reply:
Yes, and also `avg_fee_rate`, `max_fee_rate`, `min_fee_rate`, `total_fee`, `subsidy` and `total_out`.

Note that unlike most other RPCs in bitcoind, the feerates here are represented in sat/vB rather than BTC/KB, while amounts are in satoshis rather than BTC.